### ISSUE TITLE
Need to programmatically add XML mappings to existing class (in Shale we programmatically update `xml_mappings`)

### ISSUE THREAD

**Issue by kwkwan** [lutaml/lutaml-model#70]
When I ran `bundle exec rake` in  the branch `migrate-lutaml-model` of the repo `lutaml/xmi`.  I got the following error.

The problem may due to the methods to access `xml_mappings`.  It may due to the changes of  the mechanism to handle it. 

```
  1) Xmi::Sparx::SparxRoot.parse_xml loading EA CityGML extension on demand after loading extension should contains original xml mapping
     Failure/Error: expect(Xmi::Sparx::SparxRoot.mappings_for(:xml).elements).to have_key(element_key)
       expected [#<Lutaml::Model::XmlMappingRule:0x0000000102e06070 @name="Extension", @to=:extension, @render_nil=fa...ild_mappings=nil, @namespace="http://www.sparxsystems.com/profiles/CityGML/1.0", @prefix="CityGML">] to respond to `has_key?`
```

**Comment by ronaldtse** [comment]
The PR is here:
* https://github.com/lutaml/xmi/pull/49

**Comment by ronaldtse** [comment]
@kwkwan the internals of lutaml-model is very different from Shale.

Can you explain what exactly you want to do in this use case with some code? Thanks.

**Comment by ronaldtse** [comment]
@HassanAkbar this is blocking @kwkwan 's work on migrating away from Shale. Can you please help? Thanks.

**Comment by ronaldtse** [comment]
The use case is described as follows.

1. There are XML files (called "MDG files") that defines XML content structure (e.g. `CityGML_MDG_Technology.xml`).
2. There is a class `EaRoot` that inherits from Lutaml::Model::Serializable that parses an XML with a root tag `<EaRoot>`. This XML content has content that is "extensible" by the MDG files, i.e. if you have an MDG file, you extend the content allowed in `<EaRoot>`.
3. The XML file `<EaRoot>` needs to be parsed by the `EaRoot` class. This means the `EaRoot` class needs to be dynamically extended to:
    * parse new XML elements as defined by the MDG files
    * cast these new XML elements  into different dynamically generated Lutaml::Model classes.
4. Currently, the `lutaml/xmi` gem dynamically hacks `xml_mappings` within the `EaRoot` class to extend it to support those dynamically-generated classes.

This is a valid use case and we should make it easy for the user to adopt. i.e., we don't want the user to hack our internal mapping (e.g. hacking `xml_mappings` in Shale), but provide an official mechanism to do this.

**Comment by HassanAkbar** [comment]
@ronaldtse I'm looking into it, will let you know if I have any questions.

**Comment by kwkwan** [comment]
Hi @HassanAkbar Two issues (https://github.com/lutaml/lutaml-model/issues/77 and https://github.com/lutaml/lutaml-model/issues/78) are related to this issue.  I guess this issue may able be fixed if those two issues are fixed.

**Comment by ronaldtse** [comment]
@kwkwan these two issues have been fixed. Can you check if this issue can be closed? Thanks!

**Comment by kwkwan** [comment]
Closed #70 as completed at https://github.com/lutaml/lutaml-model/issues/77 and https://github.com/lutaml/lutaml-model/issues/78.
